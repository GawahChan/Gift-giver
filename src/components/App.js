import React, { Component } from 'react';
import { Button } from 'react-bootstrap';
import Gift from './Gift';
import { max_number } from '../helper';

class App extends Component {
    constructor() {
        super();
        this.state = {
            gifts: []
        }
    }

    addGift = () => {
        const { gifts } = this.state;

        gifts.push({ id: max_number(this.state.gifts.map(gift => gift.id)) + 1 }); 

        this.setState({ gifts });
    }

    removeGift = (id) => {
        const gifts = this.state.gifts.filter( gift => gift.id !== id);

        this.setState({ gifts });
    }

    render() {
        return (
            <div className='App'>
                <h1>Gift Giver</h1>
                <p>Forever keeping track of your gift list and making sure you don't forget your loved ones</p>
                <div className='gift-list'>
                    {
                        this.state.gifts.map(gift => {
                            return (
                                <Gift 
                                    key={gift.id}
                                    gift={gift}
                                    removeGift={this.removeGift}
                                />
                            );
                        })
                    }
                </div>
                <Button className='btn-add' onClick={this.addGift}>Add Gift</Button>
            </div>
        );
    }
}

export default App;
