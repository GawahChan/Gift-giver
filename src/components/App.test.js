import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

import renderer from 'react-test-renderer';

const appRendered = renderer.create(<App />).toJSON();
const app = shallow(<App />)

it('renders without crashing', () => {
    expect(appRendered).toMatchSnapshot();
});

it('initializes the `state` with an empty list of gifts', () => {
    expect(app.state().gifts).toEqual([]);
});

it('adds a new gift to `state` when clicking the `add gift` button', () => {
    app.find('.btn-add').simulate('click');

    expect(app.state().gifts).toEqual([{id: 1}]);
});