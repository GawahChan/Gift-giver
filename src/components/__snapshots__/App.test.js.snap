// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <h2>
    Gift Giver
  </h2>
  <button
    className="btn-add btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Add Gift
  </button>
</div>
`;
